driver-core/src/main/com/mongodb/client/model/WindowedComputations.java

```diff
@@ -0,0 +1,767 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.mongodb.client.model;
+
+import com.mongodb.client.model.Windows.Bound;
+import com.mongodb.lang.Nullable;
+import org.bson.BsonDocument;
+import org.bson.BsonDocumentWriter;
+import org.bson.BsonType;
+import org.bson.codecs.configuration.CodecRegistry;
+import org.bson.conversions.Bson;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+import static com.mongodb.assertions.Assertions.isTrueArgument;
+import static com.mongodb.client.model.MongoTimeUnit.MONTH;
+import static com.mongodb.client.model.MongoTimeUnit.YEAR;
+import static org.bson.assertions.Assertions.notNull;
+
+/**
+ * Builders for {@linkplain WindowedComputation windowed computations} used in the
+ * {@link Aggregates#setWindowFields(Object, Bson, List) $setWindowFields} pipeline stage
+ * of an aggregation pipeline. Each windowed computation is a triple:
+ * <ul>
+ *     <li>A window function. Some functions require documents in a window to be sorted
```

Comment:
While the whole world and the internal server docs use the word "window function", the official MongoDB docs refer to it as "[window operator](https://docs-mongodbcom-staging.corp.mongodb.com/docs/docsworker-xlarge/window-functions/reference/operator/aggregation/setWindowFields/#window-operators)". I mentioned this name in the docs of the method `Aggregates.setWindowFields`, but continued using "window function" in other places. Please tell if you think this is wrong and I should replace "function" with "operator" everywhere.